Multichain-proposal.md

```diff
@@ -0,0 +1,112 @@
+The SMPC wallet will store the cardano native tokens bridged to ethereum. SMPC network will mint the wrapped tokens. The only scripts needed are related to the minting policy for those wrapped tokens bridged from other chains. This minting policy would simply check the signature of the SMPC wallet.
+
+Tokens on different chains can be bridged to Cardano, but let’s simplify it to the scenario between Ethereum and Cardano.
+
+## Bringing Ethereum tokens into Cardano
+
+### **Sending ETH native tokens to Cardano**
+
+The bridge receives tokens from Ethereum chain by minting a *wrapped* version of them in Cardano blockchain. The minting is performed by the SMPC wallet using the single-issuer policy.
+
+Flow description:
+
+- User sends **k** *tokens E* using the ETH side of the bridge.
+- SMPC nodes build and sign a Cardano transaction that **mints** k *wrapped tokens E* and sends them to the user wallet.
+
+### Sending back wrapped ETH tokens from Cardano to Ethereum
+
+The user sends their wrapped tokens to the SMPC wallet. The corresponding ETH tokens are transferred to the user wallet on the Ethereum side of the bridge. The wrapped tokens will be burnt when there is incoming transaction from Ethereum. 
+
+Flow description:
+
+- User sends **k** *wrapped E* to the SMPC wallet.
+- SMPC nodes send **k** *tokens E* to the user wallet on the ETH side.
+- k *wrapped tokens E will be burnt when there is incoming transaction.*
+
+## Bringing Cardano tokens into Ethereum
+
+### **Sending Cardano tokens to Ethereum**
+
+The user sends their cardano native tokens to the SMPC wallet, and then the SMPC nodes send the corresponding wrapped ETH tokens on the Ethereum side. The cardano native tokens keep locked until another operation needs to receive the tokens back from Ethereum. 
+
+Flow description:
+
+- User sends **k** *tokens C* to the SMPC wallet.
+- SMPC nodes send **k** *wrapped tokens C* to the user wallet on the ETH side.
+
+### Sending back wrapped Cardano tokens from Ethereum to Cardano
+
+The user sends their wrapped tokens using the ETH side of the bridge. In the Cardano side, the SMPC wallet must have locked at least the number of tokens transferred on the ETH side. This is an invariant that must hold: *every wrapped version of a cardano token in the ETH side must have the corresponding cardano token locked in the SMPC wallet.*
+Then the corresponding amount of cardano tokens is sent to the user wallet. 
+
+Flow description:
+
+- User burn **k** *wrapped C* in the ETH side.
+- SMPC nodes build and sign a cardano transaction that sends k *tokens C* to the user wallet.
+
+# Other Specification
+
+1, Decimal
+
+Because the value of token in Cardano is limited to int 64, we need to consider the decimal convert in the 3 categories. 
+
+a, If the total supply can be expressed by int 64 with the default decimal. The wrapped tokens on Cardano will keep the default decimal as native token.
```

Comment:
What do you mean by "default decimal"?

Reply:
Default decimal here means the decimal of the token on the source chain.

Reply:
Let's change it then, "default" is confusing.